### ISSUE TITLE
1.4 vary size of hidden state in LSTM for BFs

### ISSUE THREAD

**Issue by NickleDave** [yardencsGitHub/tweetynet#68]
- [ ] re-run experiments but with LSTM layers removed?

> 1.4. Relative influence of sequential versus local features. As I understand it, TN takes as input spectrograms in which each column of the spectrogram (corresponding to some small period of time) is a "bin", and a set of labels for each bin that are derived from how the corresponding segments of song were labeled by manual annotation. The network is then trained on "windows" That contain some large number (hundreds) of contiguous bins and their labels corresponding to many hundreds of milliseconds of song. I would appreciate some guidance regarding how performance depends on the relative influence of local acoustic structure (within a single bin, or within a single syllable) and more global sequential structure that depends on the specific sequence of syllables within-training windows. I assume that both of these will affect the labels that are applied to bins, but I have little intuition about their relative influences. Will this tend to work against the correct labeling of rare variants? For example if the sequence 'abc' is very prevalent in a song, and the sequence 'abb' is very rare, will the labeling of 'abb' be biased towards the labels 'abc'? More generally, it would be helpful to have a more pointed discussion of the degree to which TN performance depends on choices for parameters such as bin size and window size - is this something that a user might want to differently adjust for BF versus canary song versus other types of vocalizations?

**Comment by NickleDave** [comment]
instead of "ablating" LSTM, run experiments where we vary size of hidden state, i.e. number of hidden units, after fixing #70 

this will be one way of getting at "Relative influence of sequential versus local features" as cited above in 1.4, although reviewer is asking about bin size v window size

**Comment by NickleDave** [comment]
based on email discussions with @yardencsGitHub after further analysis of results in initial submission:
- [ ] we will want to start with a hidden size of 1024, which is what we used for the submission
- [ ] probably experiments should be a range of halvings and doublings: 256, 512, 1024, 2048

**Comment by yardencsGitHub** [comment]
Depending on results we might want to ablate too

**Comment by NickleDave** [comment]
correction: after #75 and #76 realize that we used hidden size of 256 for initial submission.
1024 is "hidden size * 4", one dimension of the size of the learnable input-hidden weights i_h -- see LSTM variables in PyTorch docs: https://pytorch.org/docs/stable/generated/torch.nn.LSTM.html

**Comment by NickleDave** [comment]
Will need to re-run this
